src/windows/mod.rs

```diff
@@ -0,0 +1,390 @@
+pub struct Segment<'a> {
+    section: &'a IMAGE_SECTION_HEADER,
+    phantom: PhantomData<&'a SharedLibrary<'a>>,
+}
+
+impl<'a> fmt::Debug for Segment<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("Segment")
+            .field("name", &self.name())
+            .field("is_code", &self.is_code())
+            .finish()
+    }
+}
+
+impl<'a> SegmentTrait for Segment<'a> {
+    type SharedLibrary = ::windows::SharedLibrary<'a>;
+
+    #[inline]
+    fn name(&self) -> &str {
+        std::str::from_utf8(&self.section.Name).unwrap_or("").trim_end_matches('\0')
+    }
+
+    fn is_code(&self) -> bool {
+        self.name() == ".text"
+    }
+
+    #[inline]
+    fn stated_virtual_memory_address(&self) -> Svma {
+        Svma(self.section.VirtualAddress as *const u8)
+    }
+
+    #[inline]
+    fn len(&self) -> usize {
+        self.section.SizeOfRawData as usize
+    }
+}
+
+/// An iterator over Mach-O segments.
+pub struct SegmentIter<'a> {
+    sections: &'a [IMAGE_SECTION_HEADER],
+    phantom: PhantomData<&'a SharedLibrary<'a>>,
+}
+
+impl<'a> fmt::Debug for SegmentIter<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("SegmentIter").finish()
+    }
+}
+
+impl<'a> Iterator for SegmentIter<'a> {
+    type Item = Segment<'a>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.sections.is_empty() {
+            None
+        } else {
+            let section = &self.sections[0];
+            self.sections = &self.sections[1..];
+            Some(Segment {
+                section,
+                phantom: PhantomData,
+            })
+        }
+    }
+}
+
+#[repr(C)]
+struct CodeViewRecord70 {
+    signature: u32,
+    pdb_signature: GUID,
+    pdb_age: u32,
+    pdb_filename: [u8; 1],
+}
+
+/// The fallback implementation of the [SharedLibrary
+/// trait](../trait.SharedLibrary.html).
+pub struct SharedLibrary<'a> {
+    module_info: MODULEINFO,
+    module_name: OsString,
+    phantom: PhantomData<&'a SharedLibrary<'a>>,
+}
+
+impl<'a> fmt::Debug for SharedLibrary<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("SharedLibrary")
+            .field("module_base", &self.module_base())
+            .field("name", &self.name())
+            .field("debug_name", &self.debug_name())
+            .field("id", &self.id())
+            .field("debug_id", &self.debug_id())
+            .finish()
+    }
+}
+
+impl<'a> SharedLibrary<'a> {
+    fn new(module_info: MODULEINFO, module_name: OsString) -> SharedLibrary<'a> {
+        SharedLibrary {
+            module_info,
+            module_name,
+            phantom: PhantomData,
+        }
+    }
+
+    #[inline]
+    fn module_base(&self) -> *const c_char {
+        self.module_info.lpBaseOfDll as *const c_char
+    }
+
+    fn dos_header(&self) -> Option<&IMAGE_DOS_HEADER> {
+        let header: &IMAGE_DOS_HEADER = unsafe { mem::transmute(self.module_base()) };
```

Comment:
I think it would be better to use casts instead of transmutes. It's harder to get casts wrong. (Here and elsewhere.)